openedx/adg/lms/applications/tests/test_views.py

```diff
@@ -416,232 +436,419 @@ def test_get_redirects_without_login_for_education_experience_view():
     mock_business_lines.return_value = 'business_lines'
+    user = business_line_interest_view_get_request.user
 
-    if is_user_application_saved:
-        UserApplicationFactory(user=cover_letter_view_get_request.user)
-        form = 'form'
-    else:
-        form = None
+    if has_user_application:
+        UserApplicationFactory(user=user)
+        if added_education:
+            EducationFactory(user_application=user.application)
+        user.application.is_work_experience_not_applicable = added_work_exp
+        user.application.background_question = TEST_BACKGROUND_QUESTION if bg_question else ''
 
-    CoverLetterView.as_view()(cover_letter_view_get_request)
+    response = BusinessLineInterestView.as_view()(business_line_interest_view_get_request)
 
-    expected_context = {
+    context = {
         'business_lines': 'business_lines',
-        'application_form': form
+        'application_form': 'form'
     }
 
-    mock_render.assert_called_once_with(
-        cover_letter_view_get_request, 'adg/lms/applications/cover_letter.html', expected_context
-    )
-
-
-@pytest.mark.django_db
-@mock.patch('openedx.adg.lms.applications.views.render')
-@mock.patch('openedx.adg.lms.applications.views.BusinessLine.objects.all')
-@mock.patch('openedx.adg.lms.applications.views.UserApplicationCoverLetterForm')
-def test_post_with_business_line_cover_letter_view(
-    mock_cover_letter_form, mock_business_lines, mock_render, cover_letter_view_post_request
+    if will_redirect:
+        assert response.get('Location') == reverse('application_hub')
+        assert response.status_code == HTTP_302_FOUND
+    else:
+        mock_render.assert_called_once_with(
+            business_line_interest_view_get_request, 'adg/lms/applications/business_line_interest.html', context
+        )
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('has_user_application, added_work_exp, added_education, bg_question', [
+    (False, True, True, True),
+    (True, False, True, True),
+    (True, True, False, True),
+    (True, True, True, False),
+    (True, False, False, True),
+    (True, True, False, False),
+    (True, False, True, False),
+], ids=[
+    'no_application',
+    'application_with_no_work',
+    'application_with_no_education',
+    'application_with_no_bg_question',
+    'application_with_no_work_no_education',
+    'application_with_no_education_no_bg_question',
+    'application_with_no_work_no_bg_question',
+]
+)
+def test_post_without_required_prereqs_completed_business_line_interest_view(
+    logged_in_client,
+    user,
+    has_user_application,
+    added_work_exp,
+    added_education,
+    bg_question,
 ):
-    # pylint: disable=protected-access
     """
-    Test the case when the user has selected no business line
+    Tests that if a user tries to do a post request to the business line page with incomplete prereqs
+    e.g no education added, no work added etc, the user must be returned with status 400.
```

Comment:
Lets also mention background as prerequisite

Reply:
etc basically implies that. Our prerequisities might increase with the addition of references and so on